core/rawdb/database.go

```diff
@@ -238,6 +240,43 @@ func NewLevelDBDatabaseWithFreezer(file string, cache int, handles int, freezer
 	return frdb, nil
 }
 
+type counter uint64
+
+func (c counter) String() string {
+	return fmt.Sprintf("%d", c)
+}
+
+// stat stores sizes and count for a parameter
+type stat struct {
+	size  common.StorageSize
+	count counter
+}
+
+// Add size to the stat and increase the counter by 1
+func (s *stat) Add(size common.StorageSize) {
+	s.size += size
+	s.count++
+}
+
+func (s *stat) Size() string {
+	return s.size.String()
+}
+
+func (s *stat) Count() string {
+	return s.count.String()
+}
+
+// countReceiptsRLP counts how many receipts are stored in a RLP raw value
+func countReceiptsRLP(data rlp.RawValue) counter {
+	// Convert the receipts from their storage form to their internal representation
+	storageReceipts := []*types.ReceiptForStorage{}
+	if err := rlp.DecodeBytes(data, &storageReceipts); err != nil {
```

Comment:
RLP.Decode all the receipts in the database is expensive I think. Maybe we can apply some additional tricks here.
In RLP, we can count the elements in the list instead of decoding all bytes. You can check the example here https://github.com/ethereum/go-ethereum/blob/master/core/rawdb/chain_iterator.go#L140

cc @holiman Do you agree with me that using the RLP tricks here for receipts counting?

Reply:
Man, it's very much faster! :racehorse: :racehorse: :racehorse: 

Before:

`INFO [08-28|05:33:59.162] Inspection completed                     elapsed=1m6.007s`

After:

`INFO [08-28|05:50:24.877] Inspection completed                     elapsed=20.951s`